app/controllers/tasks_controller.rb

```diff
@@ -31,12 +35,40 @@ def create
     end
   end
 
+  def edit
+    @task 
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to tasks_path, notice: 'Task was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def toggle_status
```

Comment:
I think in here you should
- create a controller: TaskStatusesController
- write function: update to instead of this

Reply:
done